test/AllowListStorageReader.test.ts

```diff
@@ -0,0 +1,84 @@
+import { expect } from "chai";
+import { BigNumber, Contract } from "ethers";
+import { ethers, waffle } from "hardhat";
+
+const readStorage = async (
+  base: Contract,
+  reader: Contract,
+  method: string,
+  parameters?: (string | number | BigNumber)[],
+) => {
+  const encodedCall = reader.interface.encodeFunctionData(
+    method,
+    parameters || [],
+  );
+  const resultBytes = await base.callStatic.simulateDelegatecall(
+    reader.address,
+    encodedCall,
+  );
+  return reader.interface.decodeFunctionResult(method, resultBytes)[0];
+};
+
+describe("GPv2AllowListAuthentication", () => {
+  const [deployer, owner, solver] = waffle.provider.getWallets();
+  let authenticator: Contract;
+  let reader: Contract;
+
+  beforeEach(async () => {
+    const GPv2AllowListAuthentication = await ethers.getContractFactory(
+      "GPv2AllowListAuthentication",
+      deployer,
+    );
+    const AllowListStorageReader = await ethers.getContractFactory(
+      "AllowListStorageReader",
+      deployer,
+    );
+
+    reader = await AllowListStorageReader.deploy();
+    authenticator = await GPv2AllowListAuthentication.deploy(owner.address);
+  });
+
+  describe("getSolverAt(uint256)", () => {
```

Comment:
Nice! I like this better than just `"getSolverAt"` like we use in the other tests. Maybe its a nice follow up PR to change the other tests to use this scheme as well?